src/reversediffx.jl

```diff
@@ -0,0 +1,582 @@
+module ReverseDiffX
+
+# A lot of this module is adapted from Tracker.jl and ReverseDiff.jl
+# ReverseDiff.jl is not actively developed but it would be nice to move the code in this 
+# module to ReverseDiff at some point.
+
+export NotTracked
+
+using MacroTools, LinearAlgebra, ..ReverseDiff, StaticArrays
+using Base.Broadcast: BroadcastStyle, ArrayStyle, Broadcasted, broadcasted
+using ForwardDiff: ForwardDiff, Dual
+using ..ReverseDiff: SpecialInstruction, value, value!, deriv, track, record!, tape, unseed!
+using ..DistributionsAD: DistributionsAD, _turing_chol
+
+import SpecialFunctions, NaNMath, Zygote
+import ..DistributionsAD: turing_chol
+import Base.Broadcast: materialize
+
+const RTR = ReverseDiff.TrackedReal
+const RTV = ReverseDiff.TrackedVector
+const RTM = ReverseDiff.TrackedMatrix
+const RTA = ReverseDiff.TrackedArray
+const RDBroadcasted{F, T} = Broadcasted{<:Any, <:Any, F, T}
+
+"""
+    f(x) = dot(x, x)
+    f(x::ReverseDiff.TrackedVector) = ReverseDiff.track(f, x)
+    ReverseDiff.@grad function f(x)
+        xv = ReverseDiff.value(x)
+        return dot(xv, xv), ∇ -> (∇ * 2 * xv,)
+    end
+The `@grad` macro provides a way for the users to define custom adjoints for single-output functions wrt to their input numbers or arrays.
+"""
+macro grad(expr)
+    if @capture(expr, 
+        (f_(xs__) where {T__} = body_) | 
+        (f_(xs__) = body_) | 
+        (function f_(xs__) body_ end) | 
+        (function f_(xs__) where {T__} body_ end)
+    )
```

Comment:
Wouldn't it be easier to directly use `MacroTools.splitdef`?

Reply:
I guess that would allow to remove also other things such as `getargs_expr`.

Reply:
Possibly, I will try.